src/compiler/parser.ts

```diff
@@ -2191,17 +2195,22 @@ namespace ts {
             return allowIdentifierNames ? parseIdentifierName() : parseIdentifier();
         }
 
-        function parseTemplateExpression(): TemplateExpression {
+        function parseNoSubstitutionTemplate() {

```

Comment:
You should call this `parseTaggedNoSubstitutionTemplate` since this is otherwise very misleading. You can potentially just inline this as (reScanTemplateHeadOrNoSubstitutionTemplate(), parseLiteralNode())